logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/UpdateConversationReadDateUseCase.kt

```diff
@@ -68,11 +68,18 @@ class UpdateConversationReadDateUseCase internal constructor(
      * @param conversationId The conversation id to update the last read date.
      * @param time The last read date to update.
      */
-    operator fun invoke(conversationId: QualifiedID, time: Instant) {
-        workQueue.enqueue(ConversationTimeEventInput(conversationId, time), worker)
+    operator fun invoke(
+        conversationId: QualifiedID,
+        time: Instant,
+        shouldWaitUntilLive: Boolean = true
```

Comment:
not sure if this is the right way to solve it, since this use case will send an encrypted message, and messages should only be sent if the app is live, to avoid breaking the session

Reply:
maybe the better approach is to handle it the same way we do when FCM message is received, start incremental sync and send the message when we are live then drop after ?

Reply:
I don't see any reason to redo the sync for nothing because the app is already up to date, all messages are synced after getting the event through FCM. 
The user in this case just need to reply from notification without waiting for app to complete sync.

Reply:
but user might reply from notification with some delay, and during that delay some changes (new messages) might happen. For example: user receive notification -> lose connection for 5 minutes -> some messages comes to that conversation, but there is no connection, so the client doesn't know about it -> client goes online and sends reply instantly without sync